dgl_sparse/src/sparse_matrix.cc

```diff
@@ -167,6 +167,35 @@ c10::intrusive_ptr<SparseMatrix> SparseMatrix::RangeSelect(
   }
 }
 
+c10::intrusive_ptr<SparseMatrix> SparseMatrix::Sample(
+    int64_t dim, int64_t fanout, torch::Tensor ids, bool replace, bool bias) {
+  bool rowwise = dim == 0;
+  auto id_array = TorchTensorToDGLArray(ids);
+  auto csr = rowwise ? this->CSRPtr() : this->CSCPtr();
+
+  auto prob_or_mask =
+      bias ? TorchTensorToDGLArray(this->value()) : dgl::aten::NullArray();
+  auto sample_coo = dgl::aten::CSRRowWiseSampling(
+      CSRToOldDGLCSR(csr), id_array, fanout, prob_or_mask, replace);
+  auto sample_csr = COOToCSR(sample_coo);
```

Comment:
Is it better to do the slicing and then sampling? Then you don't need to convert sampled COO to CSR.

Reply:
I have a question regarding sampling. If we sample duplicate IDs, should the results for the same row always be the same? For instance, if we sample 2 elements from a row ```[1, 2, 3]``` three times, can the result be ```[1, 2], [2, 3], [1, 3]```? Alternatively, should it always be like ```[1, 2], [1, 2], [1, 2]```?

If we use slicing after sampling, the sampled rows from one row will always be the same. However, if we use sampling after slicing, the answers can differ.